### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by fxzjshm**
Fixes compile error on PHP, AS3 and Dart.

### CODE REVIEW

File: jtransc-gen-php/src/com/jtransc/gen/php/PhpTarget.kt [hunk 1]
```diff
@@ -413,7 +413,7 @@ class PhpGenerator(injector: Injector) : CommonGenerator(injector) {
 
 	override fun getClassInterfaces(clazz: AstClass): List<FqName> = clazz.implementingNormalized
 
-	val AstType.targetNameNullable get() = if (this.isReference()) "?$targetName" else targetName
+	val AstType.targetNameNullable get() = if (this.isReference()) "&$targetName" else targetName
```
Comment (soywiz) [inline jtransc-gen-php/src/com/jtransc/gen/php/PhpTarget.kt]: I think this is for typehinting so ? was not a typo. & is for varables to create a reference in php. In this case reference means a java type by reference, classes and arrays (not a primitive). Jtransc requires php 7.1 (dont remember if this works on hhvm in the end).

Comment (soywiz) [inline jtransc-gen-php/src/com/jtransc/gen/php/PhpTarget.kt]: http://php.net/manual/en/migration71.new-features.php

File: jtransc-rt/src/java/lang/StringBuilder.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 	"public function setStr(str:String) { this.str2 = str; buffer2 = new MyStringBuf(); buffer2.add(str); return this; }",
 })
 @JTranscAddMembers(target = "as3", value = "public var _str: String = '';")
-@JTranscAddMembers(target = "dart", value = "StringBuffer _buffer = new StringBuffer();")
+@JTranscAddMembers(target = "dart", value = "StringBuffer __buffer = new StringBuffer();")
```
Comment (soywiz) [inline jtransc-rt/src/java/lang/StringBuilder.java]: What's the rationale for this change? Im not against if it is required, but just curious.

File: jtransc-gen-php/src/com/jtransc/gen/php/PhpTarget.kt (hunk 1 above)
Comment (fxzjshm) [inline jtransc-gen-php/src/com/jtransc/gen/php/PhpTarget.kt]: Well, I'll upgrade my php and test if I can, maybe tomorrow.

File: jtransc-rt/src/java/lang/StringBuilder.java (hunk 2 above)
Comment (fxzjshm) [inline jtransc-rt/src/java/lang/StringBuilder.java]: This conflicts with the field char[] buffer ( renamed to _buffer after generation. )

**Comment by soywiz** [comment]
Also, it seems that it fails to compile: https://travis-ci.org/jtransc/jtransc/builds/254723460#L3150

If you are using intellij, since there are sometimes conflicts with standard RT, sometimes it fails to compile. What I do is to close opened RT files with errors, then `Build -> Build project` and then I can run tests.
